src/language/jhi-translate.directive.ts

```diff
@@ -50,14 +54,23 @@ export class JhiTranslateDirective implements OnChanges, OnInit {
         }
     }
 
+    ngOnDestroy() {
+        this.directiveDestroyed.next();
+        this.directiveDestroyed.complete();
+
+    }
+
     private getTranslation() {
-        this.translateService.get(this.jhiTranslate, this.translateValues).subscribe(
-            value => {
-                this.el.nativeElement.innerHTML = value;
-            },
-            () => {
-                return `${this.configService.getConfig().noi18nMessage}[${this.jhiTranslate}]`;
-            }
-        );
+        this.translateService
+            .get(this.jhiTranslate, this.translateValues)
+            .pipe(takeUntil(this.directiveDestroyed))
```

Comment:
Memory leak is only in `onLangChange` subscription. But no harm to add this safeguard here. Should we leave or remove this change?

Reply:
Yeah, I only glanced at the implementation of `TranslateService.get`. I think it might immediately complete the Observable after emitting the value, so this might not be needed. Better safe than sorry, I guess.

I can remove the change, though.

Reply:
OK, let's leave this safeguard here.

Reply:
Until someone complains that we are burning too many CPU cycles with this additional check ;)

Reply:
@vishal423, what do you think about this just in case safeguard (in the current version of `ngx-translate` this is not needed)? Keep it or not?